contexts/NotificationsContext.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, ReactNode, useState } from "react";
+import { AddNotificationFn, NotificationT, RemoveNotificationFn } from "types";
+
+export interface NotificationsContextState {
+  notifications: NotificationT[];
+  addNotification: (notification: NotificationT) => void;
+  addSuccessNotification: AddNotificationFn;
+  addErrorNotification: AddNotificationFn;
+  addPendingNotification: AddNotificationFn;
+  removeNotification: RemoveNotificationFn;
+  clearNotifications: () => void;
+}
+
+export const defaultNotificationsContextState: NotificationsContextState = {
+  notifications: [],
+  addNotification: () => null,
+  addSuccessNotification: () => null,
+  addErrorNotification: () => null,
+  addPendingNotification: () => null,
+  removeNotification: () => null,
+  clearNotifications: () => null,
+};
+
+export const NotificationsContext = createContext(defaultNotificationsContextState);
+
+export function NotificationsProvider({ children }: { children: ReactNode }) {
+  const [notifications, setNotifications] = useState<NotificationT[]>([]);
+
+  function addNotification(notification: NotificationT) {
+    setNotifications((prev) => [...prev, notification]);
+  }
+
+  function clearNotifications() {
+    setNotifications([]);
+  }
+
+  function removeNotification(transactionHash: string) {
```

Comment:
I had originally intended for this to be used exclusively for on chain interaction notifications, since we've pushed the other notification features to v2.1. I could make this more generic, but that would involve more work up front. I figured we might as well go with the minimal solution that works for now, and we can refactor to be more general when the time comes.